Common/src/main/java/me/egg82/tfaplus/utils/ConfigUtil.java

```diff
@@ -0,0 +1,127 @@
+import ninja.egg82.service.ServiceNotFoundException;
+import ninja.egg82.sql.SQL;
+import ninja.leaping.configurate.ConfigurationNode;
+import redis.clients.jedis.JedisPool;
+
+import javax.annotation.Nullable;
+import java.util.Optional;
+
+public class ConfigUtil {
+    /**
+     * Grabs the config instance from ServiceLocator
+     * @return instance of the Configuration class, may return null
+     */
+    public static @Nullable Configuration getConfig()
+    {
+        Configuration config;
+
+        try {
+            config = ServiceLocator.get(Configuration.class);
+            return config;
+        } catch (IllegalAccessException | InstantiationException | ServiceNotFoundException ex) {
+            TFAAPI.getLogger().error(ex.getMessage(), ex);
+        }
+
+        return null;
+    }
+
+    /**
+     * Grabs the cached config instance from ServiceLocator
+     * @return instance of the CachedConfigValues class, may return null
+     */
+    public static @Nullable CachedConfigValues getCachedConfig()
+    {
+        CachedConfigValues cachedConfig;
+
+        try {
+            cachedConfig = ServiceLocator.get(CachedConfigValues.class);
+            return cachedConfig;
+        } catch (IllegalAccessException | InstantiationException | ServiceNotFoundException ex) {
+            TFAAPI.getLogger().error(ex.getMessage(), ex);
+        }
+
+        return null;
+    }
+
+    public static @Nullable SQL getSQL()
+    {
+        return getCachedConfig().getSQL();
+    }
+
+    public static Tokens getTokens()
+    {
+        return getCachedConfig().getAuthy().get().getTokens();
+    }
+
+    public static boolean isDebugging() {
+        return getCachedConfig().getDebug();
+    }
+
+    public static long getIPTime() {
+        return getCachedConfig().getIPTime();
+    }
+
+    public static long getVerificationTime() {
+        return getCachedConfig().getVerificationTime();
+    }
+
+    public static ImmutableSet<String> getCommands() {
+        return getCachedConfig().getCommands();
+    }
+
+    public static boolean getForceAuth() {
+        return getCachedConfig().getForceAuth();
+    }
+
+    public static long getMaxAttempts() {
+        return getCachedConfig().getMaxAttempts();
+    }
+
+    public static FreezeConfigContainer getFreeze() {
+        return getCachedConfig().getFreeze();
+    }
+
+    public static ImmutableSet<String> getIgnored() {
+        return getCachedConfig().getIgnored();
+    }
+
+    public static JedisPool getRedisPool() {
+        return getCachedConfig().getRedisPool();
+    }
+
+    public static ConnectionFactory getRabbitConnectionFactory() {
+        return getCachedConfig().getRabbitConnectionFactory();
+    }
+
+    public static SQLType getSQLType() {
+        return getCachedConfig().getSQLType();
+    }
+
+    public static Optional<AuthyApiClient> getAuthy() {
+        return getCachedConfig().getAuthy();
+    }
+
+    public static CachedConfigValues.Builder builder() {
```

Comment:
No real reason for this, as the only time a builder is needed the ServiceLocator is updated anyway. This class wouldn't be used for that.

Reply:
I generated delegates for all CachedConfigValues in case you wanted to use things I weren't using. Feel free to remove.